### DISCUSSION TITLE
Lexicaps - Whisper Diarization - no third-paryt packages

### DISCUSSION THREAD

**Discussion by Majdoddin** [openai/whisper#1537] [Show and tell]
## [www.lexicaps.com](http://www.lexicaps.com/)
Augmenting Whisper's transcription by adding "speaker tags", so you know who says what. Currently it works for 2 speakers, and is tested for English.
I trained a classifier on top of Whisper model features (medium.en), that identifies any two speakers. _No third-party package is used for Diarization._
Seamlessly integrated with Whisper, it provides a comprehensive Transcription-Diarization service.
Give it a try or show a [Sample](http://majdoddin.github.io/call_lexicap.html).
Discussions: [Majdoddin/lexicaps](github.com/Majdoddin/lexicaps)

Thanks @karpathy for the [fun project](https://twitter.com/RMajdoddin/status/1681951115254833152?s=20)
Thanks @sidhantls for inspirative [repo](https://github.com/sidhantls/lexpod-speaker-prediction)

**Comment by databill86** [comment]
Great work! 
Are you planning to publish the source code ? :)

**Reply by ChipNowacek** [reply]
[lexicaps.com](https://lexicaps.com/) -> "We are maintaining our server. We will be back soon."
https://github.com/openai/whisper/discussions/github.com/Majdoddin/lexicaps -> 404 (seems reasonable given the URL)
https://github.com/Majdoddin/lexicaps -> just what's on this page (above)

Throw me a bone?

**Comment by freshpearYoon** [comment]
hi, thanks for great work! Is there any plan to publish source code?

**Comment by sfc-gh-praj** [comment]
@Majdoddin : this is very much needed, please let me know if you are planning to publish your code..
